internal/pkg/manifest/worker_svc_test.go

```
@@ -746,3 +746,56 @@ func TestWorkerSvc_ApplyEnv_CountOverrides(t *testing.T) {
 		})
 	}
 }
+
+type testFIFO struct {
+	FIFO *FIFOOrBool `yaml:"fifo"`
+}
+
+func Test_UnmarshalFifo(t *testing.T) {
+	testCases := map[string]struct {
+		manifest []byte
+		want     testFIFO
+		wantErr  string
+	}{
+		"fifo specified": {
```

We should also add a test case that shows how we error out if `fifo` gets an invalid input (neither `fifo` nor `enabled`). This will help us maintain and understand the behaviors & catch any future errors if the codebase is changed.